privacy-on-beam/pbeam/distinct_id.go

```diff
@@ -106,11 +117,28 @@ func DistinctPrivacyID(s beam.Scope, pcol PrivatePCollection, params DistinctPri
 	// done, remove the keys and count how many times each value appears.
 	values := beam.DropKey(s, decoded)
 	dummyCounts := beam.ParDo(s, addOneValueFn, values)
+
+	return distinctIdWithCorrectPartitions(s, partitions, epsilon, delta, maxPartitionsContributed, noiseKind, dummyCounts)
+}
+
+func distinctIdWithCorrectPartitions(s beam.Scope, partitions []beam.PCollection, epsilon, delta float64,
+	maxPartitionsContributed int64, noiseKind noise.Kind, dummyCounts beam.PCollection) beam.PCollection {
+	if len(partitions) == 0 {
```

Comment:
Similarly, it would be cleaner to do this before calling distinctIdWithCorrectPartitions